test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,215 @@
+		g.By("generating the commatrix")
+
+		cs, err := client.New()
+		if err != nil {
+			panic(fmt.Errorf("failed creating the k8s client: %w", err))
+		}
+
+		deploymentstr := "deployment"
+		isSNO, err := isSNOCluster()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deploymentstr = "sno"
+		}
+		deployment, _ := types.GetDeployment(deploymentstr)
+
+		epExporter, err := endpointslices.New(cs)
+		if err != nil {
+			panic(fmt.Errorf("failed creating the endpointslices exporter: %w", err))
+		}
+
+		commMatrix, err := commatrixcreator.New(epExporter, "", "", types.Baremetal, deployment)
+		if err != nil {
+			panic(fmt.Errorf("failed creating comm matrix creator: %w", err))
+		}
+
+		matrix, err := commMatrix.CreateEndpointMatrix()
+		if err != nil {
+			panic(fmt.Errorf("failed creating endpoint matrix: %w", err))
+		}
+
+		masterMat, workerMat := separateMatrixByRole(*matrix)
+
+		masterNFT, err := masterMat.ToNFTables()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		workerNFT, err := workerMat.ToNFTables()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("apply firewall on master nodes")
+		err = applyRulesToNode(cs, "master", masterNFT)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		if !isSNO {
+			g.By("apply firewall on worker nodes")
+			err = applyRulesToNode(cs, "worker", workerNFT)
+			o.Expect(err).ToNot(o.HaveOccurred())
+		}
+	})
+})
+
+func applyRulesToNode(cs *client.ClientSet, role string, NFTtable []byte) error {
+	utilsHelpers := utils.New(cs)
+	ns := consts.DefaultDebugNamespace + "firewall"
+	err := utilsHelpers.CreateNamespace(ns)
+	if err != nil {
+		return err
+	}
+
+	defer func() error {
+		err := utilsHelpers.DeleteNamespace(ns)
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	artifactsDir := filepath.Join(exutil.ArtifactDirPath(), "commatrix")
+	err = os.MkdirAll(artifactsDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create artifactsDir %w", err)
+	}
+
+	for _, node := range nodesList.Items {
+		if types.GetNodeRoles(&node)[0] != role {
+			continue
+		}
+
+		debugPod, err := utilsHelpers.CreatePodOnNode(node.Name, ns, consts.DefaultDebugPodImage)
+		if err != nil {
+			return fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+		}
+
+		defer func() {
+			err := utilsHelpers.DeletePod(debugPod)
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := []string{
+			"bash", "-c",
+			fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
```

Comment:
Maybe you should use an image that has nftables

Reply:
mm i dont know how to find really
we can create ours that have the image and then to install to it the nftables
u have an idea from where i can find?